tests/extras/datasets/pandas/test_xml_dataset.py

```diff
@@ -0,0 +1,328 @@
+        assert not xml_data_set.exists()
+        xml_data_set.save(dummy_dataframe)
+        assert xml_data_set.exists()
+
+    @pytest.mark.parametrize(
+        "load_args", [{"k1": "v1", "index": "value"}], indirect=True
+    )
+    def test_load_extra_params(self, xml_data_set, load_args):
+        """Test overriding the default load arguments."""
+        for key, value in load_args.items():
+            assert xml_data_set._load_args[key] == value
+
+    @pytest.mark.parametrize(
+        "save_args", [{"k1": "v1", "index": "value"}], indirect=True
+    )
+    def test_save_extra_params(self, xml_data_set, save_args):
+        """Test overriding the default save arguments."""
+        for key, value in save_args.items():
+            assert xml_data_set._save_args[key] == value
+
+    @pytest.mark.parametrize(
+        "load_args,save_args",
+        [
+            ({"storage_options": {"a": "b"}}, {}),
+            ({}, {"storage_options": {"a": "b"}}),
+            ({"storage_options": {"a": "b"}}, {"storage_options": {"x": "y"}}),
+        ],
+    )
+    def test_storage_options_dropped(self, load_args, save_args, caplog, tmp_path):
+        filepath = str(tmp_path / "test.xml")
+
+        ds = XMLDataSet(filepath=filepath, load_args=load_args, save_args=save_args)
+
+        records = [r for r in caplog.records if r.levelname == "WARNING"]
+        expected_log_message = (
+            f"Dropping `storage_options` for {filepath}, "
+            f"please specify them under `fs_args` or `credentials`."
+        )
+        assert records[0].getMessage() == expected_log_message
+        assert "storage_options" not in ds._save_args
+        assert "storage_options" not in ds._load_args
+
+    def test_load_missing_file(self, xml_data_set):
+        """Check the error when trying to load missing file."""
+        pattern = r"Failed while loading data from data set XMLDataSet\(.*\)"
+        with pytest.raises(DataSetError, match=pattern):
+            xml_data_set.load()
+
+    @pytest.mark.parametrize(
+        "filepath,instance_type,credentials",
+        [
+            ("s3://bucket/file.xml", S3FileSystem, {}),
+            ("file:///tmp/test.xml", LocalFileSystem, {}),
+            ("/tmp/test.xml", LocalFileSystem, {}),
+            ("gcs://bucket/file.xml", GCSFileSystem, {}),
+            ("https://example.com/file.xml", HTTPFileSystem, {}),
+            (
+                "abfs://bucket/file.xml",
+                AzureBlobFileSystem,
+                {"account_name": "test", "account_key": "test"},
+            ),
+        ],
+    )
+    def test_protocol_usage(self, filepath, instance_type, credentials):
+        data_set = XMLDataSet(filepath=filepath, credentials=credentials)
+        assert isinstance(data_set._fs, instance_type)
+
+        path = filepath.split(PROTOCOL_DELIMITER, 1)[-1]
+
+        assert str(data_set._filepath) == path
+        assert isinstance(data_set._filepath, PurePosixPath)
```

Comment:
we don't call `.load` here which is the one test where we have non-local protocols

Reply:
Thanks for the explanation! I see why XMLDataSet is not covered for these lines, but based on your explanation, I would expect CSVDataSet to fail as well? No? As mentioned, its mainly copy paste what i did. Anyway, I switched to tests based off JSONDataSet :) .

Reply:
The difference is that `CSVDataSet` doesn't have that early return clause for `protocol == "file://"` and so the other tests for `CSVDataSet` that do call `.load` end up covering all lines of the `load` function for CSVDataSet. But in your case, all your other tests are calling `.load` on a test file which is local, and so your `load` function does the early return and never touches the other load

Reply:
Indeed - I was looking at `CSVDataSet` in `master` and not `develop`, but both options have the special handling for local files. My bad, but looks like you've sorted it by looking at the `JSONDataset` anyway :D